engine/src/main/java/org/hibernate/validator/constraintvalidation/HibernateConstraintValidatorContext.java

```diff
@@ -107,4 +109,17 @@
 	 */
 	HibernateConstraintValidatorContext withDynamicPayload(Object payload);
 
+	/**
+	 * Returns an instance of the specified type or {@code null} if the current constraint validator payload isn't of
+	 * the given type.
+	 *
+	 * @param type the type of payload to retrieve
+	 * @return an instance of the specified type or {@code null} if the current constraint validator payload isn't of
+	 * the given type
+	 *
+	 * @since 6.0.9
+	 * @see HibernateValidatorConfiguration#constraintValidatorPayload(Object)
+	 */
+	<C> C getConstraintValidatorPayload(Class<C> type);
```

Comment:
We need the `@Incubating` annotation here. In case we want to move it elsewhere in the next version ;).

Reply:
@gsmet Added!